actors/evm/src/interpreter/instructions/lifecycle.rs

```diff
@@ -1,16 +1,129 @@
-    _state: &mut ExecutionState,
-    _platform: &'r System<'r, BS, RT>,
-    _create2: bool,
-) -> Result<(), StatusCode> {
+    state: &mut ExecutionState,
+    platform: &'r mut System<'r, BS, RT>,
+    create2: bool,
+) -> Result<U256, StatusCode> {
+    const CREATE_METHOD_NUM: u64 = 2;
+    const CREATE2_METHOD_NUM: u64 = 3;
+
+    let ExecutionState { stack, memory, .. } = state;
+    // TODO: readonly state things
+
+    // create2
+    let ret: Result<RawBytes, ActorError> = if create2 {
+        #[derive(Serialize_tuple, Deserialize_tuple)]
+        struct Create2Params {
+            #[serde(with = "strict_bytes")]
+            code: Vec<u8>,
+            salt: [u8; 32],
+        }
+
+        let endowment = stack.pop();
+        let (offset, size) = (stack.pop(), stack.pop());
+        let salt = stack.pop();
+
+        let endowment = TokenAmount::from(&endowment);
+        let input_region =
+            get_memory_region(memory, offset, size).map_err(|_| StatusCode::InvalidMemoryAccess)?;
+
+        // BE encoded array
+        let salt: [u8; 32] = salt.into();
+
+        let input_data = if let Some(MemoryRegion { offset, size }) = input_region {
+            &memory[offset..][..size.get()]
+        } else {
+            return Err(StatusCode::ActorError(ActorError::illegal_argument(
+                "initcode not in memory range".to_string(),
+            )));
+        };
+
+        // call into Ethereum Address Manager to make the new account
+        let params = Create2Params { code: input_data.to_vec(), salt };
+
+        platform.rt.send(
+            &EAM_ACTOR_ADDR,
+            CREATE2_METHOD_NUM,
+            RawBytes::serialize(&params)?,
+            endowment,
+        )
+        // errs
+    } else {
+        // create1
+        #[derive(Serialize_tuple, Deserialize_tuple)]
+        struct CreateParams {
+            #[serde(with = "strict_bytes")]
+            code: Vec<u8>,
+            nonce: u64,
+        }
+
+        let value = stack.pop();
+        let (offset, size) = (stack.pop(), stack.pop());
+
+        let value = TokenAmount::from(&value);
+        let input_region =
+            get_memory_region(memory, offset, size).map_err(|_| StatusCode::InvalidMemoryAccess)?;
+
+        let input_data = if let Some(MemoryRegion { offset, size }) = input_region {
+            &memory[offset..][..size.get()]
+        } else {
+            return Err(StatusCode::ActorError(ActorError::assertion_failed(
+                "inicode not in memory range".to_string(),
+            )));
+        };
+
+        // call into Ethereum Address Manager to make the new account
+        let params = CreateParams { code: input_data.to_vec(), nonce: state.nonce };
+        
+        platform.rt.send(&EAM_ACTOR_ADDR, CREATE_METHOD_NUM, RawBytes::serialize(&params)?, value)
+    };
+
+    // bump nonce
+    state.nonce += 1;
+    // flush nonce change 
+    platform.flush_state().unwrap();
```

Comment:
We need to actually update the state-tree in the actor state, then set the actor root.

Reply:
this is done with `rt.transaction()` correct?